pkg/object/ingresscontroller/translator.go

```diff
@@ -366,11 +370,23 @@ func (st *specTranslator) translateIngressRules(b *httpServerSpecBuilder, ingres
 		}
 
 		if existingRule == nil {
-			b.Rules = append(b.Rules, r)
+			if len(rule.Host) > 0 {
+				if rule.Host[0] == '*' {
+					wildcardHostRules = append(wildcardHostRules, r)
+				} else {
+					hostRules = append(hostRules, r)
+				}
+			} else {
+				noHostRules = append(noHostRules, r)
+			}
```

Comment:
the new rules have not been added to `b.Rules` in this case, that's the `existingRule` may have a wrong value.